puzzle.js

```diff
@@ -282,8 +293,8 @@ function PuzzleEntry(p) {
         else if (e.keyCode == 40) { this.move(e.target, 1, 0); } // down
         else if (e.keyCode == 32) { // space
             this.dx = 1 - this.dx; this.dy = 1 - this.dy;
-            if (e.currentTarget.parentElement.classList.contains("given-fill")) return;
-            if (this.options["data-fill-cycle"]) { this.currentFill = this.cycleClasses(e.target.parentElement, this.fillClasses, e.shiftKey); }
+            if (e.currentTarget.classList.contains("given-fill")) return;
```

Comment:
In the keyCode == 32 block, I found that adding

```
// Reset focus after dx, dy change.
e.target.blur();
e.target.focus();
```

Allowed the dx, dy changes to propagate to the crossword highlighting.

It seems a little hacky, so I'm open to other suggestions :)